curriculum/challenges/english/20-upcoming-python/learn-lambda-functions-by-building-a-expense-tracker/657ee2baaadeaa670bc238ac.md

```diff
@@ -0,0 +1,101 @@
+---
+id: 657ee2baaadeaa670bc238ac
+title: Step 25
+challengeType: 20
+dashedName: step-25
+---
+
+# --description--
+
+After getting the category, we filter the expenses and print the filtered list. The `filter_expenses_by_category` function returns an iterator, so we convert it to a list before printing.
+
+These two lines are already added to the code:
+
+```py
+print(f"\nExpenses for {category}:")
+expenses_from_category = filter_expenses_by_category(expenses, category)
+```
+
+- The first line prints the category heading, using an f-string to insert the user-chosen category into the message.
+- The second line calls a function to filter the expenses list, returning only those in the specified category. The result is stored in `expenses_from_category`.
+
+Using the `print_expenses` function, write the code to output the filtered expenses. Pass the `expenses_from_category` as an argument, ensuring it's converted into a list. Your task is to correctly implement this as one line in your code.
```

Comment:
```suggestion
Using the `print_expenses` function, write the code to output the filtered expenses. Pass the `expenses_from_category` as an argument, ensuring it's converted into a list. Your task is to correctly implement this as one line in your code.
```
Is conversion to `list` necessary?

Reply:
it's not necessary as the code will still run fine, but the `list` conversion will convert the filter object into a list. This is for good practice

Reply:
Using `list` to convert generator into a list, instead of less clear ways, is indeed a good practice.

I don't believe it is for a situation like this. Result of `list(expenses_from_category)` is immediately passed to the `print_expenses` which is handling generator returned from `filter` without issues. Resulting list is not assigned to any variable. This also means `expenses_from_category` generator is exhausted at that point. Using `list` is obsolete.

One could argue that somebody might want to do something else with the list of filtered expenses, but in that case `list` should be used in the following way:
```python
expenses_from_category = list(filter_expenses_by_category(expenses, category))
print_expenses(expenses_from_category)
```
This assures that after printing expenses list of the filtered expenses still can be used.

Reply:
I think this makes sense. Thanks for the insight. I will get rid of the `list` and update all the steps after this one too